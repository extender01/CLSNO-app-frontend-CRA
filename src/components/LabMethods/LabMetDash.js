import React from 'react';
import {connect} from 'react-redux';

import LabMetList from './LabMetList';
import LabMetCategory from './LabMetCategory';
import LabMetSearch from './LabMetSearch';
import {startLoadTests} from '../../actions/testActions';




class LabMetDash extends React.Component {
    constructor(props) {
        super(props);
        
        

    }

    componentDidMount() {
        // this.props.loadTests();
        

    }

    render() {
        return (
            <div className='lm__container'>
                <div className='lm'>
                    {this.props.labTestsError === null ?
                        <React.Fragment>
                            <LabMetSearch />
                            <LabMetCategory />
                            <LabMetList />
                        </React.Fragment>
                        : 
                        <h2>nelze se spojit s db</h2>}
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
        labTestsError: state.labTests.error
    };
};

const mapDispatchToProps = (dispatch) => {
    return {
        loadTests: () => {
            dispatch(startLoadTests());
        }
    };
};


export default connect(mapStateToProps, mapDispatchToProps)(LabMetDash);


