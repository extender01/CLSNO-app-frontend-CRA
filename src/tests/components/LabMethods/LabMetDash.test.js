import React from 'react';
import {shallow} from 'enzyme';
import LabMetDash from '../../../components/LabMethods/LabMetDash';


test('should render LabMetDash correctly', () => {
    const wrapper = shallow(<LabMetDash />);
    expect(wrapper).toMatchSnapshot();
});

