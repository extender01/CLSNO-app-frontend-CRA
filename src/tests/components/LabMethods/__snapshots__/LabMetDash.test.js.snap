// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LabMetDash correctly 1`] = `
<div
  className="dash"
>
  <Connect(LabMetSearch) />
  <Connect(LabMetCategory) />
  <Connect(LabMetList) />
</div>
`;
